<?xml version="1.0"?>
<launch>

    <node pkg="body_handling_demo" type="body_handling_change.py" name="body_handling_action_node" output="screen" >
        <!-- 物料框位置 -->
        <param name="cargo_box1" value="11" type="string"/>
        <param name="cargo_box2" value="12" type="string"/>
        <!-- <param name="cargo_box3" value="21" type="string"/>
        <param name="cargo_box4" value="22" type="string"/> -->
        <!-- 物料的位置 -->
        <param name="cargo1" value="101" type="string"/>
        <param name="cargo2" value="102" type="string"/>
        <param name="cargo3" value="103" type="string"/>
        <param name="cargo4" value="104" type="string"/>

        <!-- 层数->升降机高度 -->
        <param name="floor1" value="0.05" type="double"/>
        <param name="floor2" value="0.45" type="double"/>
        <param name="floor3" value="0.85" type="double"/>
        <param name="floor4" value="0.999" type="double"/>

        <!-- 物料->抓取物料名称 -->
        <param name="label1" value="YiHeDa" type="string"/>
        <param name="label2" value="JiaoHuanJi" type="string"/>
        <param name="label3" value="NSK" type="string"/>
        <param name="label4" value="AnNiu" type="string"/>

        <!-- 固定位置->升降机高度 -->
        <param name="lift_send" value="0.6" type="double"/>
        <param name="lift_table" value="0.26" type="double"/>
        
        <!-- 工具坐标系名称 -->
        <param name="left_tool1" value="grip_left" type="string"/>
        <param name="left_tool2" value="nip_left" type="string"/>
        <param name="left_tool3" value="tip_left" type="string"/>
        <param name="right_tool1" value="grip_right" type="string"/>
        <param name="right_tool2" value="nip_right" type="string"/>
        <param name="right_tool3" value="tip_right" type="string"/>

        <!-- 设置自动模式的持续时间 s-->
        <param name="auto_mode_time" value="300" type="int"/>

        <!-- 设置手动模式的持续时间 s-->
        <param name="manual_mode_time" value="86400" type="int"/>

        <!-- 设置自主充电点位 -->
        <rosparam param="charge_range">[80, 90]</rosparam>
        <param name="charge_pose" value="120" type="string"/>

        <!-- 设置头部舵机扭动角度和容错 -->
        <rosparam param="servo_motor">[347, 500]</rosparam>
        <rosparam param="servo_tolerant">[3, 3]</rosparam>

        <!-- 是否进行语音播报 -->
        <param name="talk_flag" value="true" type="bool"/>

    </node>

</launch>